widget/embedded/src/store/ui.ts

```diff
@@ -8,14 +13,44 @@ interface UiState {
   currentPage: string;
   setSelectedSwap: (requestId: string | null) => void;
   setCurrentPage: (path: string) => void;
+  isActiveTab: boolean | null;
+  showActivateTabModal: boolean;
+  activateCurrentTab: (
+    swaps: PendingSwap[],
+    setCurrentTabAsActive: () => void
+  ) => void;
+  setShowActivateTabModal: (flag: boolean) => void;
 }
-
 export const useUiStore = createSelectors(
-  create<UiState>()((set) => ({
+  create<UiState>()((set, get) => ({
     connectWalletsButtonDisabled: false,
     selectedSwapRequestId: null,
     currentPage: '',
+    isActiveTab: null,
+    showActivateTabModal: false,
     setSelectedSwap: (requestId) => set({ selectedSwapRequestId: requestId }),
     setCurrentPage: (path) => set({ currentPage: path }),
+    activateCurrentTab: (swaps, setCurrentTabAsActive) => {
+      const { showActivateTabModal } = get();
+      const anyRunningSwap = swaps.some((swap) => swap.status === 'running');
+
+      if (!showActivateTabModal && anyRunningSwap) {
+        set({ showActivateTabModal: true });
+      } else if (showActivateTabModal || !anyRunningSwap) {
+        setCurrentTabAsActive();
+      }
+    },
+    setShowActivateTabModal: (flag) => {
+      set({ showActivateTabModal: flag });
+    },
   }))
 );
+
+const tabManager: TabManagerInterface = new TabManager({
```

Comment:
Couldn't we move/restructure the TabManager as a hook like other logic aspects? 🤔

Reply:
We should maintain only a single instance of the tab manager class. In the current setup, it functions as a singleton. However, if we transition to using hooks and unintentionally employ it in various locations, multiple instances might be created. To ensure we have only one instance, we could introduce a tab manager provider and context, but this adds complexity without a clear necessity.